core/domain/voiceover_domain.py

```diff
@@ -352,3 +352,91 @@ def create_empty(
             language_accent_code=language_accent_code,
             voiceovers_mapping={},
             automated_voiceovers_audio_offsets_msecs={})
+
+
+class VoiceoverAutogenerationPolicy:
+    """A domain object for the voiceover autogeneration policy."""
+
+    def __init__(
+        self,
+        language_codes_mapping: Dict[str, Dict[str, bool]],
+        autogenerated_voiceovers_are_enabled: bool = False
+    ) -> None:
+        """Constructs an VoiceoverAutogenerationPolicy domain object.
+
+        Args:
+            language_codes_mapping: dict(str, dict(str, bool)). A
+                dict mapping language_codes to nested dicts. Each nested dict
+                contains language_accent_codes as keys and booleans indicating
+                if we can generate automatic voiceovers for the corresponding 
+                language_accent_code as values.
+            autogenerated_voiceovers_are_enabled: bool. Indicates whether 
+                cloud-based voiceover generation is enabled.
+        """
+        self.language_codes_mapping = language_codes_mapping
+        self.autogenerated_voiceovers_are_enabled = (
+            autogenerated_voiceovers_are_enabled)
+
+    def validate(self) -> None:
+        """Validates the EntityVoiceovers object."""
+        if not isinstance(self.language_codes_mapping, dict):
+            raise utils.ValidationError(
+                'language_codes_mapping must be a dict, received %s'
+                % self.language_codes_mapping)
+        for language_code in self.language_codes_mapping:
+            if not isinstance(language_code, str):
+                raise utils.ValidationError(
+                    'language_code must be a string, received %s'
+                    % language_code)
+            if not utils.is_supported_audio_language_code(language_code):
+                raise utils.ValidationError(
+                    'Invalid language_code: %s' % language_code)
```

Comment:
Make it more descriptive to clarify that we expect a supported audio language code.
```suggestion
                    'Expected language code to be in supported audio language codes, '
                    'received: %s' % language_code)
```

Reply:
Fixed: https://github.com/oppia/oppia/pull/22168#discussion_r2014547558